Mathlib/Probability/Independence/Kernel.lean

```diff
@@ -0,0 +1,916 @@
+    (hf_meas : ∀ i, Measurable (f i)) (i j k : ι) (hik : i ≠ k) (hjk : j ≠ k) :
+    IndepFunₖ (f i * f j) (f k) κ μ := by
+  have : IndepFunₖ (fun ω => (f i ω, f j ω)) (f k) κ μ :=
+    hf_Indepₖ.indepFunₖ_prod hf_meas i j k hik hjk
+  change IndepFunₖ ((fun p : β × β => p.fst * p.snd) ∘ fun ω => (f i ω, f j ω)) (id ∘ f k) κ μ
+  exact IndepFunₖ.comp this (measurable_fst.mul measurable_snd) measurable_id
+
+@[to_additive]
+theorem iIndepFunₖ.indepFunₖ_finset_prod_of_not_mem [IsMarkovKernel κ] {ι : Type _} {β : Type _}
+    {m : MeasurableSpace β} [CommMonoid β] [MeasurableMul₂ β] {f : ι → Ω → β}
+    (hf_Indepₖ : iIndepFunₖ (fun _ => m) f κ μ) (hf_meas : ∀ i, Measurable (f i))
+    {s : Finset ι} {i : ι} (hi : i ∉ s) :
+    IndepFunₖ (∏ j in s, f j) (f i) κ μ := by
+  classical
+    have h_right : f i =
+      (fun p : ∀ _j : ({i} : Finset ι), β => p ⟨i, Finset.mem_singleton_self i⟩) ∘
+      fun a (j : ({i} : Finset ι)) => f j a := rfl
+    have h_meas_right : Measurable fun p : ∀ _j : ({i} : Finset ι), β
+      => p ⟨i, Finset.mem_singleton_self i⟩ := measurable_pi_apply ⟨i, Finset.mem_singleton_self i⟩
+    have h_left : ∏ j in s, f j = (fun p : ∀ _j : s, β => ∏ j, p j) ∘ fun a (j : s) => f j a := by
+      ext1 a
+      simp only [Function.comp_apply]
+      have : (∏ j : ↥s, f (↑j) a) = (∏ j : ↥s, f ↑j) a := by rw [Finset.prod_apply]
+      rw [this, Finset.prod_coe_sort]
+    have h_meas_left : Measurable fun p : ∀ _j : s, β => ∏ j, p j :=
+      Finset.univ.measurable_prod fun (j : ↥s) (_H : j ∈ Finset.univ) => measurable_pi_apply j
+    rw [h_left, h_right]
+    exact
+      (hf_Indepₖ.indepFunₖ_finset s {i} (Finset.disjoint_singleton_left.mpr hi).symm hf_meas).comp
+        h_meas_left h_meas_right
+
+@[to_additive]
+theorem iIndepFunₖ.indepFunₖ_prod_range_succ [IsMarkovKernel κ] {β : Type _}
+    {m : MeasurableSpace β} [CommMonoid β] [MeasurableMul₂ β] {f : ℕ → Ω → β}
+    (hf_Indepₖ : iIndepFunₖ (fun _ => m) f κ μ) (hf_meas : ∀ i, Measurable (f i)) (n : ℕ) :
+    IndepFunₖ (∏ j in Finset.range n, f j) (f n) κ μ :=
+  hf_Indepₖ.indepFunₖ_finset_prod_of_not_mem hf_meas Finset.not_mem_range_self
+
+theorem iIndepSetₖ.iIndepFunₖ_indicator [Zero β] [One β] {m : MeasurableSpace β} {s : ι → Set Ω}
+    (hs : iIndepSetₖ s κ μ) :
+    iIndepFunₖ (fun _n => m) (fun n => (s n).indicator fun _ω => 1) κ μ := by
+  classical
+  rw [iIndepFunₖ_iff_measure_inter_preimage_eq_mul]
+  rintro S π _hπ
+  simp_rw [Set.indicator_const_preimage_eq_union]
+  refine' @hs S (fun i => ite (1 ∈ π i) (s i) ∅ ∪ ite ((0 : β) ∈ π i) (s i)ᶜ ∅) fun i _hi => _
+  have hsi : MeasurableSet[generateFrom {s i}] (s i) :=
+    measurableSet_generateFrom (Set.mem_singleton _)
+  refine'
+    MeasurableSet.union (MeasurableSet.ite' (fun _ => hsi) fun _ => _)
+      (MeasurableSet.ite' (fun _ => hsi.compl) fun _ => _)
+  · exact @MeasurableSet.empty _ (generateFrom {s i})
+  · exact @MeasurableSet.empty _ (generateFrom {s i})
```

Comment:
In this file, sometimes you indent things after `classical`, and sometimes you don't. Could you make a choice once and for all? (Personally, I would prefer not indenting, but it's up to you!)